kernel/kernel-impl/src/main/java/org/sakaiproject/event/impl/ClusterEventTracking.java

```diff
@@ -689,21 +690,33 @@ public Object readSqlResultRecord(ResultSet result)
 		M_log.debug("Starting (after) Event #: {}", m_lastEventSeq);
 	}
 
-    /**
-     * KNL-1184
-     * Initializes the events cache, if enabled
-     */
-    private void initCacheServer() {
-        // remove down to and including this line
-        cachingEnabled = serverConfigurationService().getBoolean("memory.cluster.enabled", false);
-        if (cachingEnabled) {
-            eventCache = memoryService().newCache("org.sakaiproject.event.impl.ClusterEventTracking.eventsCache");
-            /**
-             * This cache only needs to hold a single value, the last updated event id
-             */
-            eventLastCache = memoryService().newCache("org.sakaiproject.event.impl.ClusterEventTracking.eventLastCache");
-        }
-    }
+	/**
+	 * KNL-1184
+	 * Initializes the events cache, if enabled
+	 */
+	private void initCacheServer() {
+		// remove down to and including this line
+		cachingEnabled = serverConfigurationService().getBoolean("memory.cluster.enabled", false);
+		if (cachingEnabled) {
+			boolean eventsCachesUsed = false;
+			String[] caches = serverConfigurationService().getStrings("memory.cluster.names");
+			if(ArrayUtils.isNotEmpty(caches)) {
+				for(String cacheName : caches) {
+					if("org.sakaiproject.event.impl.ClusterEventTracking.eventsCache".equals(cacheName)) {
+						eventCache = memoryService().newCache("org.sakaiproject.event.impl.ClusterEventTracking.eventsCache");
+						eventsCachesUsed = true;
+					} else if("org.sakaiproject.event.impl.ClusterEventTracking.eventLastCache".equals(cacheName)) {
+						/**
+						 * This cache only needs to hold a single value, the last updated event id
+						 */
+						eventLastCache = memoryService().newCache("org.sakaiproject.event.impl.ClusterEventTracking.eventLastCache");
+						eventsCachesUsed = true;
+					}
+				}
+				cachingEnabled = eventsCachesUsed;
```

Comment:
Looking closely I think this can result in a NPE, as if you have a config of: 

    memory.cluster.enabled=true
    memory.cluster.names=org.sakaiproject.event.impl.ClusterEventTracking.eventsCache

then Sakai will startup and then when it attempts to call `writeEventToCluster` it will NPE as only one of the caches is used. I think you have to have all or nothing.